### ISSUE TITLE
Adding `--import-map` breaks the code

### ISSUE THREAD

**Issue by canonic-epicure** [denoland/deno#10616]
I have code that works fine without import maps:
```
nickolay@frontier:~/workspace/Bryntum/siesta-monorepo/siesta$ deno run -A --unstable --quiet tests/hook/hook.t.js
Launching test file: tests/hook/hook.t.js in Deno 1.9.2

 PASS  tests/hook/hook.t.js

Test files : 1 passed, 0 failed, 1 total
Time       : 33ms
```
Adding a dummy import map breaks it:
```
nickolay@frontier:~/workspace/Bryntum/siesta-monorepo/siesta$ deno run -A --unstable --quiet --import-map import_map.json tests/hook/hook.t.js
error: Unmapped bare specifier "path"
```

How is this possible? It seems adding a source map changes the set of files being imported somehow. I do have the bare "path" but only in files that are used for Node. And again, the code works fine w/o import maps.

Import map content for the full picture:
```
{
  "imports":  {
    "siesta/":  "./"
  }
}
```

To reproduce:
1) clone the `git@github.com:bryntum/siesta.git`, specific revision is: 6e7e2576e0afcdd3f4f6e1966923ea46c29896eb
2) `npm i` , `npx tsc`
3) `cd examples/deno`
4) Launch w/o import maps: `deno run -A --unstable --quiet --no-check tests/index.ts`
Should see something like:
```
Launching test suite project: Awesome Deno project test suite in Deno 1.9.2

 FAIL  module/zoomer.t.js
├─✘ Using `zoomer` should work
│ └─✘ equal(received, expected)
│   
│     2 | import { zoomer } from "../../src/module.js";
│     3 | it('Using `zoomer` should work', async (t) => {
│   ➤ 4 |     t.equal(zoomer('Myers Courtney', 35), { name: 'Courtney Myers', age: 35 });
│       |       ^
│     5 | });
│   
│   Received                   │ │ Expected                  
│                              │ │                           
│   {                          │ │ {                         
│     "age": 35,               │ │   "age": 35,              
│     "name": "Myers Courtney" │ │   "name": "Courtney Myers"
│   }                          │ │ }                         
│   
```
5) Launch with import maps: `deno run -A --unstable --quiet --no-check --import-map import_map.json tests/index.ts`
The output is:
```
error: Unmapped bare specifier "path"
```

**Comment by satyarohith** [comment]
The value of "siesta" should point to a js/ts file.

**Comment by lucacasonato** [comment]
Do you use any dynamic imports?

**Comment by canonic-epicure** [comment]
@satyarohith No, it can point a directory too, thats the whole point of the import maps

@lucacasonato I do, yes, all of them don't use bare specifiers though.

**Comment by lucacasonato** [comment]
Do any of the files that are dynamically imported have bare specifiers?

**Comment by canonic-epicure** [comment]
@lucacasonato Nope

**Comment by lucacasonato** [comment]
Ok let me rephrase, do any of the modules that are directly or indirectly imported through a module which was dynamically imported have bare specifiers?

**Comment by lucacasonato** [comment]
Ok let me rephrase, do any of the modules that are directly or indirectly imported through a module which was dynamically imported have bare specifiers?

**Comment by canonic-epicure** [comment]
Updated the ticket with steps to reproduce.

@lucacasonato No, currently none of the Deno-specific source files use bare specifiers. However, the Node-specific source files do. When launching in Deno, the Node-specific files are not expected to be imported, which is confirmed by the fact that code works w/o import maps. Seems that once the import map is added, somehow the Node-specific file leaks in.

The dynamic imports happens like this: https://github.com/bryntum/siesta/blob/master/src/siesta/project/Project.ts#L140
And the `isDeno` function is here: https://github.com/bryntum/siesta/blob/master/src/util/Helpers.ts#L188

Can it be that `Boolean(globalThis.Deno.build)` is false when using import map?

**Comment by lucacasonato** [comment]
> Can it be that Boolean(globalThis.Deno.build) is false when using import map?

No, but it looks like our dependency analysis is not working correctly with import maps and dynamic imports. cc @bartlomieju

**Comment by nayeemrmn** [comment]
This is a case of #10168. You have conditional imports (which won't run in Deno) with static specifiers of some Node file, and that file is being statically dep-analysed and type-checked. Only the former for JS files.

In fact the author of that issue was likely doing the same thing. Duplicate of #10168.

**Comment by canonic-epicure** [comment]
@nayeemrmn I see. Thanks for letting know, looking forward for your PR to land!

**Comment by canonic-epicure** [comment]
@nayeemrmn Realistically, how much time do you think it will take, for your PR to land? I'm planning the release of our tool and just need to decide if we skip using import maps for now, or not..
